diff --git a/src/Text/HTML/TagSoup/Entity.hs b/src/Text/HTML/TagSoup/Entity.hs
index e8a32e3..76cd0ac 100644
--- a/src/Text/HTML/TagSoup/Entity.hs
+++ b/src/Text/HTML/TagSoup/Entity.hs
@@ -67,7 +67,7 @@ lookupNamedEntity = \x -> Map.lookup x mp
 -- > escapeXML "hello & world" == "hello &amp; world"
 escapeXML :: String -> String
 escapeXML = concatMap $ \x -> IntMap.findWithDefault [x] (ord x) mp
-    where mp = IntMap.fromList [(ord b, "&"++a++";") | (a,[b]) <- ("#39","\'"):xmlEntities]
+    where mp = IntMap.fromList [(ord b, "&"++a++";") | (a,[b]) <- xmlEntities]
 
 
 -- | A table mapping XML entity names to resolved strings. All strings are a single character long.
